mock-interactions.js

```diff
@@ -8,9 +8,15 @@
  * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
  */
 
-(function(global) {
+// We declare it as a namespace to be compatible with JSCompiler.
+/** @const */ var MockInteractions = {};
+
+(function(scope, global) {
   'use strict';
 
+  // In case the var above was not global, or if it was renamed.
+  global.MockInteractions = scope;
```

Comment:
Line 12 must be in the format of var namespace = {}, so that JSCompiler recognizes it as a namespace and can infer types from it.

Line 18 was added for backwards compatibility. I saw at least one client internally that loaded this script inside an IIFE, which meant that the var was not a global variable anymore. Also JSCompiler is allowed to rename the var, but it won't touch the global.foo assignment. (Not sure if it can rename the scope.* assignments though.)

I agree that these two together seem a bit hacky. It would be somewhat simpler if this file didn't use an IIFE at all. That might still break that client I mentioned though. Maybe an `@export` could also fix it (because non-JSCompiler clients would ignore `@export`, but for JSCompiler this would generate code that put it into the global namespace regardless of how this file is loaded).